addons/sale_stock_margin/models/sale_order_line.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.depends('product_id', 'company_id', 'currency_id', 'product_uom', 'move_ids', 'move_ids.stock_valuation_layer_ids', 'order_id.picking_ids.state')
+    def _compute_purchase_price(self):
```

Comment:
you could update the original _compute_purchase_price
``` python
    @api.depends(lambda self: self._purchase_price_depends())
    def _compute_purchase_price(self):
        [...]
```

and only add the dependencies by inheriting _purchase_price_depends